src/metabase/integrations/slack.clj

```diff
@@ -36,14 +36,17 @@
   (when-not (str/blank? channel-name)
     (if (str/starts-with? channel-name "#") (subs channel-name 1) channel-name)))
 
+(defsetting slack-cache
```

Comment:
I would suggest naming this something like `cached-channel-and-user-names` for a few reasons:

1. `slack-` doesn't really give us any extra information, we know this has to do with Slack because it's in a `.slack` namespace -- see https://github.com/metabase/metabase/wiki/Metabase-Clojure-Style-Guide#dont-repeat-namespace-alias-in-function-names for more info.
2. `cache` is way to vague. `cache` _of what?_

Reply:
I think you're exactly right on 2, I should have renamed that.

Also, I need some clarification on 1. Because this value is a `defsetting`, it needs a globally unique name. And looking at other defsettings the linked section of the guide isn't being followed. Here are a few examples I found:

- [snowplow.clj#L30](https://github.com/metabase/metabase/blob/sharing-dashboard-subscriptions-slack-performance/src/metabase/analytics/snowplow.clj#L30)
- [snowplow.clj#L38](https://github.com/metabase/metabase/blob/sharing-dashboard-subscriptions-slack-performance/src/metabase/analytics/snowplow.clj#L38)
- [geojson.clj#L83](https://github.com/metabase/metabase/blob/sharing-dashboard-subscriptions-slack-performance/src/metabase/api/geojson.clj#L83)
- [geojson.clj#L83](https://github.com/metabase/metabase/blob/sharing-dashboard-subscriptions-slack-performance/src/metabase/api/geojson.clj#L83)
- [geojson.clj#L83](https://github.com/metabase/metabase/blob/sharing-dashboard-subscriptions-slack-performance/src/metabase/api/geojson.clj#L83)
- [ldap.clj#L16](https://github.com/metabase/metabase/blob/sharing-dashboard-subscriptions-slack-performance/src/metabase/integrations/ldap.clj#L16)
- [ldap.clj#L21](https://github.com/metabase/metabase/blob/sharing-dashboard-subscriptions-slack-performance/src/metabase/integrations/ldap.clj#L21)
- [ldap.clj#L24](https://github.com/metabase/metabase/blob/sharing-dashboard-subscriptions-slack-performance/src/metabase/integrations/ldap.clj#L24)
- [ldap.clj#L29](https://github.com/metabase/metabase/blob/sharing-dashboard-subscriptions-slack-performance/src/metabase/integrations/ldap.clj#L29)
- [ldap.clj#L38](https://github.com/metabase/metabase/blob/sharing-dashboard-subscriptions-slack-performance/src/metabase/integrations/ldap.clj#L38)
- [ldap.clj#L41](https://github.com/metabase/metabase/blob/sharing-dashboard-subscriptions-slack-performance/src/metabase/integrations/ldap.clj#L41)
- [slack.clj#L16](https://github.com/metabase/metabase/blob/sharing-dashboard-subscriptions-slack-performance/src/metabase/integrations/slack.clj#L16)
- [slack.clj#L22](https://github.com/metabase/metabase/blob/sharing-dashboard-subscriptions-slack-performance/src/metabase/integrations/slack.clj#L22)
- [slack.clj#L27](https://github.com/metabase/metabase/blob/sharing-dashboard-subscriptions-slack-performance/src/metabase/integrations/slack.clj#L27)

At a glance, the majority of the `defsettings` have their namespace prefixed into them. I figured the reason for this is that when reading _the code_ it's clear what namespace the var came from, but when looking at settings e.g. in the database I don't think you get the same affordance. Is that correct? I could use some more context. Maybe `slack-cached-channel-and-user-names` is the right name.